src/components/Chart/Chart.js

```diff
@@ -0,0 +1,297 @@
+        // console.log('componentDidMount');
+        this.calculateQuantiles();
+        
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        // console.log(this.props.summaryStart, this.props.summaryEnd)
+        if (prevProps.summaryStart !== this.props.summaryStart || 
+            prevProps.summaryEnd !== this.props.summaryEnd ||
+            prevProps.dataset !== this.props.dataset ||
+            prevProps.scenarios !== this.props.scenarios ||
+            prevProps.scale !== this.props.scale) {
+            this.calculateQuantiles();
+        }
+    }
+
+    calculateQuantiles = () => {
+        const { dataset, firstDate, summaryStart, summaryEnd, stat, width, height, scenarios } = this.props;
+        const { severities } = this.state;
+        const quantileObj = {}
+        
+        const startIdx = getDateIdx(firstDate, summaryStart);
+        const endIdx = getDateIdx(firstDate, summaryEnd);
+        // console.log(startIdx, endIdx);
+        let globalMaxVal = 0;
+        // console.log(scenarios)
+        // console.log(stat)
+        quantileObj[stat] = {};
+        for (let severity of severities) {
+            // console.log(severity)
+            quantileObj[stat][severity] = {};
+            for (let scenario of scenarios) {
+                // console.log(scenario)
+                // every Chart has a given stat passed down from ChartContainer
+                // every Chart will contain all scenarios and all severities
+                // startIdx and endIdx specify the time range on which we want to calc quantiles
+                // every sim.vals array will be sliced on this timeRange 
+                // then every day of a simulation will be summed up returning an Array of sim sums 
+                // then d3.quantiles can be applied to the Array to create final desired obj
+                
+                const sumArray = dataset[scenario][severity][stat].sims.map(sim => {
+                    return sim.vals.slice(startIdx, endIdx).reduce((a, b) => a + b, 0)
+                } );
+                // console.log(scenario, stat, severity, sumArray)
+                const minVal = min(sumArray)
+                const maxVal = max(sumArray)
+                const tenth = quantile(sumArray, 0.10)
+                const quartile = quantile(sumArray, 0.25)
+                const median = quantile(sumArray, 0.5)
+                const thirdquartile = quantile(sumArray, 0.75)
+                const ninetyith = quantile(sumArray, 0.9)
+                
+                // keep track of largest value across all the severities and scenarios for yAxis
+                if (maxVal > globalMaxVal) globalMaxVal = maxVal
+
+                quantileObj[stat][severity][scenario] = {
+                    minVal, maxVal, tenth, quartile, median, thirdquartile, ninetyith, //xScale, yScale
+                }
+            }
+        }
+        
+        console.log(quantileObj)
+        let yScale;
```

Comment:
`const` are nice because you know for sure the assignment of this variable cannot change. You don't currently have a strong type check for your project but if you did, you would run into issue where `yScale` is mutable and may not have a guaranteed type after checking it. It also nice to know when reading code that a variable always has the same assignment vs `let` where it would indicate it could have changed somewhere. 

```
const yScale = this.props.scale === 'linear' ?
    scaleLinear().range([height - margin.bottom, margin.top]).domain([0, globalMaxVal])
    : yScale = scalePow().exponent(0.25).range([height - margin.bottom, margin.chartTop]).domain([0, globalMaxVal])
```

Though until you get a type checker, it's not a big deal.